owlbot.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This script is used to synthesize generated parts of this library."""
+
+import logging
+from pathlib import Path
+import shutil
+import subprocess
+
+import os
+import synthtool as s
+from synthtool.languages import php
+from synthtool import _tracked_paths
+
+logging.basicConfig(level=logging.DEBUG)
+
+# (dirname, version)
+protos = [
+    ("api", "api"),
+    ("extendedoperations", "cloud"),
+    ("location", "cloud"),
+    ("logging", "google"), # for the metadata
+    ("logging", "cloud"),
+    ("iam", "google"), # for the metadata
+    ("iam", "cloud"),
+    ("iamlogging", "iam"),
+    ("rpc", "rpc"),
+    ("type", "type"),
+]
```

Comment:
FYI @chingor13 @SurferJeffAtGoogle I am doing some unusual things with `version` and the copy paths here. I am not sure how else to do this, but if there's a better way please let me know!